js/model/Presentation.js

```diff
@@ -357,7 +357,17 @@ export var Presentation = {
 
     get title() {
         var svgTitles = this.document.root.getElementsByTagNameNS(SVG_NS, "title");
-        return svgTitles.length ? svgTitles[0].firstChild.wholeText.trim() : "Untitled";
+        if (svgTitles.length) {
+            return svgTitles[0].firstChild.wholeText.trim();
+        }
+        else {
+            var svgRoot = this.document.root.getElementsByTagNameNS(SVG_NS, "svg");
+            if (svgRoot.length) {
+                var docName = svgRoot.firstChild.getAttributeNode("sodipodi:docname").value.trim();
```

Comment:
> Line 370 will return "Untitled" if the element attribute does not exist. (at least in theory)

As I understand it, line 370 is executed if `svgTitles.length` and `svgRoot.length` are both equal to zero.